keyboards/monsgeek/m1/m1.c

```diff
@@ -0,0 +1,628 @@
+static uint8_t  RGB_HSV_level;
+
+HSV hsv;
+
+void led_test(uint8_t color);
+void clear_eeprom(void);
+void rgb_hsv_updata_user(void);
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+    if (!process_record_user(keycode, record)) {
+        return false;
+    }
+    switch (keycode) {
+        case MO(WIN_FN):
+        case MO(MAC_FN):
+            fn_make_flag = record->event.pressed;
+            return true;
+        case KC_GRV:
+            if (fn_make_flag && record->event.pressed) {
+                Lkey_flag    = true;
+                current_time = timer_read();
+                scancode     = KC_GRV;
+                return false;
+            } else {
+                Lkey_flag = 0;
+            }
+            return true;
+        case KC_LCTL:
+            if (fn_make_flag && record->event.pressed) {
+                if (while_test_flag) {
+                    while_test_flag = false;
+                    rgb_matrix_init();
+                } else {
+                    Lkey_flag    = true;
+                    current_time = timer_read();
+                    scancode     = KC_LCTL;
+                }
+                return false;
+            } else {
+                Lkey_flag = 0;
+            }
+            return true;
+        case DF(WIN_B):
+            if ((fn_make_flag && record->event.pressed) && (alarm_flag == 0)) {
+                alarm_flag = true;
+                rgb_matrix_toggle_noeeprom();
+                current_time = timer_read();
+                set_single_persistent_default_layer(WIN_B);
+                layer_state_set(0x0);
+                return false;
+            }
+            return true;
+        case DF(MAC_B):
+            if ((fn_make_flag && record->event.pressed) && (alarm_flag == 0)) {
+                alarm_flag = true;
+                rgb_matrix_toggle_noeeprom();
+                current_time = timer_read();
+                set_single_persistent_default_layer(MAC_B);
+                keymap_config.no_gui     = 0;
+                eeconfig_update_keymap(keymap_config.raw);
+                return false;
+            }
+            return true;
+        case TG(WIN_W):
+            if ((fn_make_flag && record->event.pressed) && (alarm_flag == 0)) {
+                alarm_flag = true;
+                rgb_matrix_toggle_noeeprom();
+                current_time = timer_read();
+            }
+            return true;
+        case TG(MAC_W):
+            if ((fn_make_flag && record->event.pressed) && (alarm_flag == 0)) {
+                alarm_flag = true;
+                rgb_matrix_toggle_noeeprom();
+                current_time = timer_read();
+            }
+            return true;
+        case GU_TOGG:
+            if ((fn_make_flag && record->event.pressed) && (alarm_flag == 0)) {
+                alarm_flag = true;
+                rgb_matrix_toggle_noeeprom();
+                current_time = timer_read();
+            }
+            return true;
```

Comment:
One concern I have here is that there is no way to disable or change this functionality.  Users will be locked into this, or forced to fully replace and reimplement the functionality of these keycodes just to get them to behave normally. 

Wrapping them with a `#ifndef DISABLE_M1_KEYCODES` or similar would allow users to disable the functionality, if they desire without reimplementing.